ui/src/app/settings/selectors/general/general.js

```diff
@@ -0,0 +1,46 @@
+const general = {};
+
+/**
+ * Returns object with osinfo data
+ * @param {Object} state - the redux state
+ * @returns {Object} The osinfo data
+ */
+general.osinfo = state => state.general.items;
+
+/**
+ * Returns true if general is loading
+ * @param {Object} state - the redux state
+ * @returns {Boolean} - general is loading
+ */
+general.loading = state => state.general.loading;
+
+/**
+ * Returns true if general is loaded
+ * @param {Object} state - the redux state
+ * @returns {Boolean} - general is loading
+ */
+general.loaded = state => state.general.loaded;
+
+/**
+ * Returns kernels data
+ * @param {Object} state - the redux state
+ * @returns {Array} the available kernel options
+ */
+general.getKernalOptionsByKey = (state, key) => {
```

Comment:
What is `key` here? I think it would be beneficial to give it a more descriptive name and include it in the doc block above.

Reply:
Fixed